code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -19,110 +18,112 @@
 	                            /obj/item/weapon/implantcase/chem,
 	                            /obj/item/weapon/reagent_containers/pill/time_release)
 
-	afterattack(obj/target, mob/user , flag)
-		if(!user.Adjacent(target))
-			return
-
-		if(!target.reagents)
-			if(filled)
-				if(istype(target, /obj/machinery/artifact))
-					src.reagents.clear_reagents()
-					user << "<span class='notice'>You squirt the solution onto the [target]!</span>"
-					filled = 0
-					icon_state = "dropper[filled]"
-			return
-
-		if(filled)
-
-			if(target.reagents.total_volume >= target.reagents.maximum_volume)
-				user << "<span class='warning'>[target] is full.</span>"
-				return
-
-			if(!target.is_open_container() && !ismob(target) && !is_type_in_list(target, injectable_types)) //You can inject humans and food but you cant remove the shit.
-				user << "<span class='warning'>You cannot directly fill this object.</span>"
-				return
-
-			var/trans = 0
-
-			if(ismob(target))
-				if(istype(target , /mob/living/carbon/human))
-					var/mob/living/carbon/human/victim = target
-
-					var/obj/item/safe_thing = victim.get_body_part_coverage(EYES)
-
-					if(safe_thing)
-						if(!safe_thing.reagents)
-							safe_thing.create_reagents(100)
-						trans = src.reagents.trans_to(safe_thing, amount_per_transfer_from_this)
-
-						for(var/mob/O in viewers(world.view, user))
-							O.show_message(text("<span class='danger'>[] tries to squirt something into []'s eyes, but fails!</span>", user, target), 1)
-						spawn(5)
-							src.reagents.reaction(safe_thing, TOUCH)
-
-
-
-						user << "<span class='notice'>You transfer [trans] units of the solution.</span>"
-						if (src.reagents.total_volume<=0)
-							filled = 0
-							icon_state = "dropper[filled]"
-						return
+/obj/item/weapon/reagent_containers/dropper/update_icon()
+	icon_state = "dropper[(reagents.total_volume ? 1 : 0)]"
 
+/obj/item/weapon/reagent_containers/dropper/afterattack(obj/target, mob/user , flag)
+	if(!user.Adjacent(target))

```

Comment:
See my above comment.

You should always keep the same argument names, no exceptions.